internal/controlplane/controller.go

```diff
@@ -0,0 +1,100 @@
+package controlplane
+
+//need to parse flags
+//configs for gw and ingress
+//start prom collector
+//start prom http server
+//start controller
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+var (
+	testScheme = runtime.NewScheme()
+)
+
+func init() {
+	// Register core Kubernetes types
+	utilruntime.Must(clientgoscheme.AddToScheme(testScheme))
+
+	// Register Gateway API types
+	utilruntime.Must(gatewayv1.AddToScheme(testScheme))
+	utilruntime.Must(gatewayv1beta1.AddToScheme(testScheme))
```

Comment:
more for my curiosity, why we need the beta APIs? I though the idea was to start just with the GA ones

Reply:
yea, good point, ill remove it